core/src/main/scala-2.12/cats/compat/FoldableCompat.scala

```diff
@@ -0,0 +1,22 @@
+package cats
+package compat
+
+private[cats] trait FoldableCompat[F[_]] { this: Foldable[F] =>
```

Comment:
if this is `private[cats]` can people override outside of cats? I don't know how this works when it is mixed into `Foldble`. Is it even publicly visible at all?

Reply:
I think this is not accessible outside `cats` so `FoldableCompat` cannot be extended but `iterable` can be overridden through `Foldable`.
I ran a quick test for this extending from outside the `cats` package with
```scala
object test {

  trait T1[F[_]] extends cats.Foldable[F] {
    override def iterable[A](fa: F[A]): Stream[A] = ???
  }

  trait T2[F[_]] extends cats.compat.FoldableCompat[F] {
    override def iterable[A](fa: F[A]): Stream[A] = ???
  }
}
```
and it seems that `T1` is ok and `T2` fails as expected on 2.12 and 2.13
```
[error] /Users/fmariotti/open-source/cats/core/src/main/scala/test.scala:7:38: trait FoldableCompat in package compat cannot be accessed in package cats.compat
[error]   trait T2[F[_]] extends cats.compat.FoldableCompat[F] {
[error]                                      ^
[error] one error found
[error] (coreJVM / Compile / compileIncremental) Compilation failed
```
```
[error] /Users/fmariotti/open-source/cats/core/src/main/scala/test.scala:7:38: trait FoldableCompat in package compat cannot be accessed in package cats.compat
[error]   trait T2[F[_]] extends cats.compat.FoldableCompat[F] {
[error]                                      ^
[error] one error found
[error] (coreJVM / Compile / compileIncremental) Compilation failed
```